fornax-hea/build-sas.sh

```diff
@@ -0,0 +1,182 @@
+
+
+############# Definition of software versions #############
+ubuntu_version=24.04
+sas_version=22.1.0
+py_version=$PYTHON_VERSION
+###########################################################
+
+
+############### Setting up useful variables ###############
+# SAS gets upset if it can't find Perl - this will need to be included in an activate.d script
+#  as well, to make sure this path is set when the user loads the environment in
+export SAS_PERL=/usr/bin/perl
+# We set this to point at the conda environment we're just about to produce - ENV_DIR is not
+#  set in this script, but we can trust that it will be
+export SAS_PYTHON=$ENV_DIR/sas/bin/python
+
+# This is where we will be putting the calibration files for XMM
+export SAS_CCFPATH=$SUPPORT_DATA_DIR/xmm_ccf/
+
+##### DOWNLOADING SAS #####
+# SAS will be downloaded from HEASARC - this is the base for the populated URL
+base_sas_link=https://heasarc.gsfc.nasa.gov/FTP/xmm/software/sas/${sas_version}/Linux/Ubuntu${ubuntu_version}/
+
+# This is the more general structure of the tar file name (see it in the HEASARC FTP for other versions
+#  of SAS), but unfortunately there isn't a file with this easy to populate name structure for
+#  the Ubuntu version (24.04) that I'm working with right now
+# sas_file=sas_${sas_version}-Ubuntu${ubuntu_version}.tgz
+
+# HARD CODED - this is the non-generalised file link to the tar of SAS I'm working with right now (22.1.0 for Ubuntu 24.04)
+sas_file=sas_${sas_version}-a8f2c2afa-20250304-ubuntu${ubuntu_version}-gcc13.3.0-x86_64.tgz
+# HARD CODED - This is the non-generalised name of the directory that XMM-SAS gets 'built' in
+sas_install_dir=xmmsas_22.1.0-a8f2c2afa-20250304
+
+# Assembling the download link
+sas_link=$base_sas_link${sas_file}
+###########################
+
+
+###########################################################
+
+########### Download and unpack required files ############
+wget $sas_link \
+	&& tar xvf $sas_file \
+	&& rm -f $sas_file 
+###########################################################
+
+
+############# Setup the SAS Conda environment #############
+# Creates a Conda definition file that can be used to setup the environment that SAS 
+#  will be associated with
+# We use the Conda-hosted HEASoft to avoid downloading and building HEASoft manually
+cat <<EOF > conda-sas.yml
+name: sas
+channels:
+  - https://heasarc.gsfc.nasa.gov/FTP/software/conda
+  - conda-forge
+dependencies:
+  - python=$py_version
+  - heasoft=6.35.*
+  - pytest
+  - rsync
+EOF
+
+# Use the yml to create the SAS env
+bash /usr/local/bin/conda-env-install.sh
+
+# Use the Python requirements file included in the SAS directory to install
+#  the module that it wants for some included Python bits (e.g. pySAS)
+# Other build scripts for Fornax images seem to use the uv package manager, so we 
+#  will as well - particularly the pip CLI version
+micromamba run -n sas uv pip install -r sas_python_packages.txt
```

Comment:
I don't think uv should be here. This is adding to a conda environment, so pip will do.

Also, please clean the cache after the call, or use `--no-cache-dir` so the cache files don't stay in the image